app/controllers/user.rb

```diff
@@ -0,0 +1,32 @@
+get '/users' do
+  @user = User.all
+  erb :index
+end
+
+get '/users/new' do
+  @user = User.new
+  erb :'users/new'
+end
+
+get '/users/:id' do
+  @user = User.find_by(id: params[:id])
+  redirect '/' unless @user.id = session[:user_id]
```

Comment:
👍  protecting the serving of the user profile

Reply:
Use your current_user helper here though, and everywhere else you ask the session a question:
```ruby
redirect '/' unless @user == current_user
```